components/resolved_ts/src/endpoint.rs

```diff
@@ -313,12 +356,11 @@ where
             }),
             before_start: None,
             on_error: Some(Box::new(move |observe_id, _region, e| {
-                let error = e.extract_error_header();
                 scheduler_error
-                    .schedule(Task::RegionError {
+                    .schedule(Task::ReRegisterRegion {
```

Comment:
Do not re-register region if we met errors like `ReadIndexNotReady`.

Reply:
Yes, but we still need a way to retry though, perhaps we should do the retry in `get_snapshot`, and leave `on_error` as an interface for error that can't handle by retry.